base/ltcounts.dtx

```diff
@@ -305,16 +305,29 @@
 %  \begin{macro}{\@definecounter}
 %  \changes{v1.1b}{1995/05/20}{Streamlined code}
 %  \changes{v1.1c}{1995/05/20}{And do it right}
+%  \changes{v1.1n}{2023/11/07}{Do not change \cs{the...} if already defined (gh/823)}
 %
 %    \begin{macrocode}
 \def\@definecounter#1{\expandafter\newcount\csname c@#1\endcsname
      \setcounter{#1}\z@
      \global\expandafter\let\csname cl@#1\endcsname\@empty
      \@addtoreset{#1}{@ckpt}%
      \global\expandafter\let\csname p@#1\endcsname\@empty
+%    \end{macrocode}
+%    If \cs{the\#1} is undefined or \cs{relax} we define it with the
+%    standard definition for counters, otherwise we warn. This will
+%    catch, for example, that somebody defines a counter named
+%    ``index'' conflicting with the \env{theindex} environment.
+%    \begin{macrocode}
      \expandafter
-     \gdef\csname the#1\expandafter\endcsname\expandafter
-          {\expandafter\@arabic\csname c@#1\endcsname}}
+     \ifx\csname the#1\endcsname\relax
```

Comment:
Are we really worried about `\relax` here? Why not just `\ifcsname`?

Reply:
because it would be wrong. Packages, e.g., bigfoot, touch the counter name and as a result it is defined but as \relax. The above is just the normal way of LaTeX to determine if a command is free, ie either undefined or \relax and given that in this case we define it, it doesn't matter that we test with \ifx

Reply:
I was thinking that if the counter has ended up as `\relax` its not really 'in use' ... and I guess with my `expl3` view of the word, it's really not defined at all ;)

Assuming you want to stick this way, why are you not using `\@ifundefined`?

Reply:
it is not the counter that is \relax, it is `\the<counter>` that is \relax and that is what technically happens, because packages do (for one or the other reason) `\csname the<counter>\endcsname` before the a `\newcounter{<counter>}` was issued. I started with using  `\ifcsname` and several tests blow into my face, especially those around footnotes and floats.  In the past touching `\the<counter>` didn't matter because afterwards it was unconditionally defined by the code above.

as for using `\@ifundefined` I could have used that, it is just a little bit slower as it unnecessarily first tests with \ifcsname. But it is perhaps cleaner.

Reply:
I think `\expandafter\ifx\csname` is fine here and matches the style of the rest of the block (` \expandafter
     \gdef\csname...`  etc)

Reply:
that was roughly my thinking (despite the fact that it is slightly faster)